### DISCUSSION TITLE
How to override "Running "serverless" from node_modules"?

### DISCUSSION THREAD

**Discussion by Knufle** [serverless/serverless#10876] [Q&A]
Hi there everybody, hope you're all doing fine!

This discussion is basically a more objective question regarding my previous discussion https://github.com/serverless/serverless/discussions/10840 (please check it if you want more in-depth info)

Basically what I'm trying to do is run a serverless command in a folder after removing some of its dependencies, whenever I run serverless commands before actually trimming the dependencies, I get this message:

![image](https://user-images.githubusercontent.com/25941471/159124638-02dfd75f-4485-4f2e-9fe7-66ad68eee920.png)

Which I understand happens because my node_modules has serverless in it.

So after trimming dependencies with "trim-node-modules.sh" which is basically this https://github.com/viktorfa/strapi-serverless/blob/master/scripts/trim-node-modules.sh I can't run serverless commands anymore, because it tries to use the local version.

![image](https://user-images.githubusercontent.com/25941471/159124670-ea425fdd-1ba2-4e99-acc2-e0cc6b6eaad6.png)

My question is, can I force it to use the globally installed version? If so, how?

**Comment by pgrzesik** [comment]
Hello @Knufle - there's not way to force the global version, but if you're already trimming depdendencies from local `node_modules`, you can remove `serverless` dependency as well and it will use the global version.

**Reply by Knufle** [reply]
Hi @pgrzesik, thanks for answering, I figured the guide I was following was a little bit old so the author's npm installed dependencies differently, I ended up using "npm i --legacy-peer-deps" which didn't install serverless as a local dependency, and fortunately got rid of this issue.

**Reply by pgrzesik** [reply]
Good to hear that you've been able to resolve your problem 👍

**Comment by jeanbmar** [comment]
Legacy peer deps won't work in all situations. It should be considered a bug that Serverless uses the node_modules from, let's say, a third-party plugin dev dependencies, instead of the Serverless installed globally.
